mmdet/apis/ote/apis/detection/nncf_task.py

```diff
@@ -0,0 +1,242 @@
+        optimization_type: OptimizationType,
+        dataset: Dataset,
+        output_model: ModelEntity,
+        optimization_parameters: Optional[OptimizationParameters],
+    ):
+        if optimization_type is not OptimizationType.NNCF:
+            raise RuntimeError("NNCF is the only supported optimization")
+
+        train_dataset = dataset.get_subset(Subset.TRAINING)
+        val_dataset = dataset.get_subset(Subset.VALIDATION)
+        config = self._config
+
+        if optimization_parameters is not None:
+            update_progress_callback = optimization_parameters.update_progress
+        else:
+            update_progress_callback = default_progress_callback
+        time_monitor = TrainingProgressCallback(update_progress_callback)
+        learning_curves = defaultdict(OTELoggerHook.Curve)
+        training_config = prepare_for_training(config, train_dataset, val_dataset, time_monitor, learning_curves)
+        mm_train_dataset = build_dataset(training_config.data.train)
+
+        if torch.cuda.is_available():
+            self._model.cuda(training_config.gpu_ids[0])
+
+        # Initialize NNCF parts if start from not compressed model
+        if not self._compression_ctrl:
+            self._create_compressed_model(mm_train_dataset, training_config)
+
+        # Run training.
+        self._training_work_dir = training_config.work_dir
+        self._is_training = True
+        self._model.train()
+
+        train_detector(model=self._model,
+                       dataset=mm_train_dataset,
+                       cfg=training_config,
+                       validate=True,
+                       compression_ctrl=self._compression_ctrl)
+
+        # Check for stop signal when training has stopped. If should_stop is true, training was cancelled
+        if self._should_stop:
+            logger.info('Training cancelled.')
+            self._should_stop = False
+            self._is_training = False
+            return
+
+        self.save_model(output_model)
+
+        output_model.model_status = ModelStatus.SUCCESS
+        self._is_training = False
+
+    def save_model(self, output_model: ModelEntity):
+        buffer = io.BytesIO()
+        hyperparams = self._task_environment.get_hyper_parameters(OTEDetectionConfig)
+        hyperparams_str = ids_to_strings(cfg_helper.convert(hyperparams, dict, enum_to_str=True))
+        labels = {label.name: label.color.rgb_tuple for label in self._labels}
+        modelinfo = {
+            'compression_state': self._compression_ctrl.get_compression_state(),
+            'meta': {
+                'config': self._config,
+                'nncf_enable_compression': True,
+            },
+            'model': self._model.state_dict(),
+            'config': hyperparams_str,
+            'labels': labels,
+            'VERSION': 1,
+        }
+
+        torch.save(modelinfo, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+
+    def cancel_training(self):
```

Comment:
As NNCFDetectionTask does not implement ITrainingTask interface. It should no be here. Or one should change interface and add 'cancel_optimization' for example, i don't know.

Reply:
@Ilya-Krylov `cancel_training` is one of those methods (like `IUnload`) which should have been removed from the task interface but somehow escaped scrutiny. So no need to add a `cancel_optimization`, we can delete `cancel_training` instead.